File: packages/mobile/src/types/caregiver.ts
export interface PatientSummary {
  id: string;
  first_name: string;
  last_name: string;
  email: string | null;
  phone: string;
  assigned_at: string;
  assignment_notes: string | null;
}

export interface CaregiverProfile {
  id: string;
  first_name: string;
  last_name: string;
  email: string;
  phone: string | null;
  is_active: boolean;
  created_at: string;
  assigned_patients: PatientSummary[];
}

export interface AssignPatientRequest {
  patient_id: string;
  notes?: string;
}

export interface AssignmentResponse {
  id: string;
  caregiver_id: string;
  patient_id: string;
  assigned_at: string;
  notes: string | null;
}

File: packages/backend/src/care_assistant/models/caregiver.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class CaregiverSummary(BaseModel):
    """Minimal caregiver info embedded inside patient responses."""

    id: str
    first_name: str
    last_name: str
    email: str
    phone: Optional[str] = None
    assigned_at: datetime
    assignment_notes: Optional[str] = None


class PatientSummary(BaseModel):
    """Minimal patient info embedded inside caregiver responses."""

    id: str
    first_name: str
    last_name: str
    email: Optional[str] = None
    phone: str
    assigned_at: datetime
    assignment_notes: Optional[str] = None


class CaregiverProfile(BaseModel):
    """Full caregiver profile with list of assigned patients."""

    id: str
    first_name: str
    last_name: str
    email: str
    phone: Optional[str] = None
    is_active: bool
    created_at: datetime
    assigned_patients: list[PatientSummary] = []


class AssignPatientRequest(BaseModel):
    patient_id: str
    notes: Optional[str] = None


class AssignmentResponse(BaseModel):
    id: str
    caregiver_id: str
    patient_id: str
    assigned_at: datetime
    notes: Optional[str] = None
